src/sciwrappers.c

```diff
@@ -1018,6 +1018,12 @@ void sci_scroll_columns(ScintillaObject *sci, gint columns)
 }
 
 
+void sci_scroll_lines(ScintillaObject *sci, gint lines)
+{
+	SSM(sci, SCI_LINESCROLL, 0, (uptr_t) lines);
+}
+
+
```

Comment:
I don't think it's worth adding a new Scintilla wrapper function for the single caller, given the call is pretty simple.  Just inline the equivalent in the calling code.

Reply:
Same could be said of...

https://github.com/geany/geany/blob/a44df8bb570e5d43f882649826e5d34ca3907a63/src/sciwrappers.c#L1015-L1018

Reply:
Disagree with @b4n, all Scintilla interfacing should go through wrappers so that changes Scintilla makes can be abstracted away, for example the recent string length change.

The performance difference will be immaterial unless the Scintilla call is being hammered in a loop (which it never is AFAICT).  And if it _is_ a problem anywhere, define the function `inline` in `sciwrappers.h` (C99).

Reply:
@xiota sure there might be some already very seldom used, not sure it's a trend to follow…
But if @elextr has a strong opinion, fine by me. FWIW, IMO it doesn't make much sense where it's a single cas that's just as easy to fix in the caller or refactor out when there are more users.

Reply:
@elextr I'm not concerned by any speed issues, just that although it looks "clean", it also makes the code somewhat more complex, adding a tiny bit of an additional constraints through new interfaces just to transform that interface (the wrapper doesn't have any logic of its own)

Reply:
@b4n I am normally the one complaining about single use functions where a lump of functionality is broken out for no good reason, and that case totally matches your reasoning.

But for APIs wrapping is a useful way of getting encapsulation and hiding, and the Geany plugin API should have been done that way, but well, that ship has sailed, hit stormy seas, and sunk ;-D

Probably doesn't matter at this stage which this individual call uses but every little drop of water erodes the stone [proverb].